src/components/Button/Button.tsx

```diff
@@ -1,17 +1,35 @@
 import * as PropTypes from 'prop-types'
 import * as React from 'react'
 
-import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { UIComponent, childrenExist, customPropTypes, Extendable } from '../../lib'
```

Comment:
I would expect `/lib` to be runtime utilities the library needs and `/types` to hold all our types.  That would mean moving `Extendable` to `/types` as well.

There are a few more typing utilities in `/types/theme.d.ts` (ObjectOf, OneOrArray) that we could put in a separate `/types/utils` along with this `Extendable` util.  It seems we should keep all these in one place.

Thoughts?

Reply:
totally agree 👍

Reply:
I added Extendable in the utils.d.ts file in the types and updated other components accordingly. Thanks for the suggestion!